scanner/matcher/updater/distribution/updater.go

```diff
@@ -0,0 +1,113 @@
+package distribution
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"github.com/quay/claircore"
+	"github.com/quay/zlog"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/scanner/datastore/postgres"
+	"golang.org/x/exp/slices"
+)
+
+const updateInterval = 24 * time.Hour
+
+var (
+	random = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	jitterHours = []time.Duration{
+		1 * time.Hour,
+		2 * time.Hour,
+		3 * time.Hour,
+		4 * time.Hour,
+		5 * time.Hour,
+	}
+)
+
+type Updater struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	store postgres.MatcherStore
+
+	// known tracks all known distributions.
+	known []claircore.Distribution
+	// mutex protects access to known.
+	mutex sync.RWMutex
+}
+
+func New(ctx context.Context, store postgres.MatcherStore) (*Updater, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	u := &Updater{
+		ctx:    ctx,
+		cancel: cancel,
+
+		store: store,
+	}
+	return u, nil
+}
+
+func (u *Updater) Known() []claircore.Distribution {
+	known := func() []claircore.Distribution {
+		u.mutex.RLock()
+		defer u.mutex.RUnlock()
+
+		return slices.Clone(u.known)
+	}()
+
+	return known
```

Comment:
Is wrapping this in a `func()` meant to reduce the time the lock is held?  Seems like this would be a negligible improvement in such a small function, something I'm missing?

Reply:
yeah pretty negligible lock holding reduction. I can simplify it